app/mailers/govuk_notify_mailer.rb

```diff
@@ -0,0 +1,37 @@
+# Extract to gem and inherit in the app from this class
+#
+class GovukNotifyMailer < ActionMailer::Base
+  default delivery_method: :govuk_notify
+
+  attr_accessor :govuk_notify_template
+  attr_accessor :govuk_notify_personalisation
+
+  protected
+
+  def mail(headers = {})
+    _validate!
+
+    headers[:body] ||= _default_body
+
+    message = super(headers)
+    message.govuk_notify_template = govuk_notify_template
+    message.govuk_notify_personalisation = govuk_notify_personalisation
+  end
+
+  def set_template(template)
```

Comment:
Might be useful to do type checking here and in `set_personalisation`. Not a blocker.